src/app/core/components/archive-payer/archive-payer.component.spec.ts

```diff
@@ -0,0 +1,171 @@
+/* @format */
+import { MessageService } from '../../../shared/services/message/message.service';
+import { AccountService } from '@shared/services/account/account.service';
+import { HttpClient, HttpHandler } from '@angular/common/http';
+import { ComponentFixture, TestBed } from '@angular/core/testing';
+import { ArchivePayerComponent } from './archive-payer.component';
+import { ArchiveVO, AccountVO } from '@models/index';
+
+describe('ArchivePayerComponent', () => {
+  let component: ArchivePayerComponent;
+  let fixture: ComponentFixture<ArchivePayerComponent>;
+  let mockAccountService;
+
+  beforeEach(() => {
+    mockAccountService = {
+      getAccount: jasmine
+        .createSpy('getAccount')
+        .and.returnValue({ accountId: '1' }),
+    };
+  });
+
+  beforeEach(async () => {
+    await TestBed.configureTestingModule({
+      declarations: [ArchivePayerComponent],
+      providers: [
+        { provide: AccountService, useValue: mockAccountService },
+        HttpClient,
+        HttpHandler,
+        MessageService,
+      ],
+    }).compileComponents();
+
+    fixture = TestBed.createComponent(ArchivePayerComponent);
+    component = fixture.componentInstance;
+    fixture.detectChanges();
+  });
+
+  it('should create', () => {
+    expect(component).toBeTruthy();
+  });
+
+  it('displays no payer when there is no payer id assigned to the account', () => {
+    component.hasPayer = false;
+    fixture.detectChanges();
+    const divElement = fixture.nativeElement.querySelector('#no-payer');
+    expect(divElement).toBeTruthy();
+  });
+
+  it('displays payer data when there is a payer id assigned to the account', () => {
+    const payer = new AccountVO({});
+    payer.fullName = 'Test name';
+    payer.primaryEmail = 'test@email.com';
+    payer.accessRole = 'access.role.owner';
+
+    component.payer = payer;
+    component.hasPayer = true;
+
+    fixture.detectChanges();
+    const divElement = fixture.nativeElement.querySelector('#with-payer');
+    expect(divElement).toBeTruthy();
+  });
+
+  it('displays the component even if no payer was found for unknown reasons', () => {
+    component.hasPayer = true;
+
+    fixture.detectChanges();
+    const divElement = fixture.nativeElement.querySelector('#with-payer');
+    expect(divElement).toBeTruthy();
+  });
+
+  it('displays the current payer data if the logged account is different than the payer', () => {
+    const payer = new AccountVO({});
+
+    payer.accountId = '1';
+    payer.fullName = 'Test name';
+    payer.primaryEmail = 'test@email.com';
+    payer.accessRole = 'access.role.owner';
+
+    component.hasPayer = true;
+    component.isPayerDifferentThanLoggedUser = true;
+    component.payer = payer;
+
+    fixture.detectChanges();
+    const div = fixture.nativeElement.querySelector('#is-not-payer');
+    expect(div).toBeTruthy();
+  });
+
+  it('displays the current payer data if the logged account is different than the payer', () => {
```

Comment:
This should have a different name from the previous test. It looks like this one is meant to be the case where the payer is the logged in user?